core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -192,6 +195,7 @@ public Node(Settings preparedSettings) {
         this(InternalSettingsPreparer.prepareEnvironment(preparedSettings, null), Collections.<Class<? extends Plugin>>emptyList());
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
Could you move the unchecked stuff to a smaller method? Like just make a little method that does the unchecked work so this whole thing isn't unchecked.